src/app.js

```diff
@@ -1,23 +1,154 @@
-  res.send("Dashboard");
+    const user = await User.findOne({ email });
+    if (!user) {
+      throw new Error("Invalid Credentials");
+    } else {
+      const isPasswordMatching = await bcrypt.compare(password, user.password);
+      if (!isPasswordMatching) {
+        throw new Error("Invalid Credentials");
+      } else {
+        res.send("Login successfull");
+      }
+    }
+  } catch (err) {
+    res.status(400).send({ message: err.message });
+  }
 });
 
-app.use("/hello", (req, res) => {
-  res.send("Hello world");
+//GET - feed
+app.get("/feed", async (req, res, next) => {
+  try {
+    const users = await User.find({});
+    if (!users.length) {
+      res.status(404).send("No users");
+    } else {
+      res.send(users);
+    }
+  } catch (err) {
+    res.status(400).send({ message: err.message });
+  }
 });
 
-app.use("/test", (req, res) => {
-  // request handler function
-  res.send("hello from the server");
+// GET - user
+app.get("/user", async (req, res, next) => {
+  const userEmail = req.body.email;
+
+  try {
+    const users = await User.find({ email: userEmail });
+    if (!users.length) {
+      res.status(404).send("User not found");
+    } else {
+      res.send(users);
+    }
+  } catch (err) {
+    res.status(400).send({ message: err.message });
+  }
 });
 
-const PORT = 3030;
-app.listen(PORT, () => {
-  console.log("Server is successfully listening on port", PORT);
+// GET - firstUser
+app.get("/firstUser", async (req, res, next) => {
+  try {
+    const user = await User.findOne({ email: req.body.email });
+    if (!user) {
+      res.status(404).send("User not found");
+    } else {
+      res.send(user);
+    }
+  } catch (err) {
+    res.status(400).send({ message: err.message });
+  }
 });
+
+// DELETE
+app.delete("/user", async (req, res, next) => {
+  try {
+    const deletedUser = await User.findByIdAndDelete(req.body.userID);
+    if (!deletedUser) {
+      res.status(404).send("User not found");
+    } else {
+      res.send({ message: "User deleted successfully", data: deletedUser });
+    }
+  } catch (err) {
+    res.status(400).send({ message: err.message });
+  }
+});
+
+// UPDATE
+app.patch("/user/:userID", async (req, res, next) => {
+  try {
+    const userID = req.params?.userID;
+    const updateBody = req.body;
+
+    // adding check for
+    const allowedUpdate = ["firstName", "lastName", "gender", "age", "skills"];
+
+    const isUpdateAllowed = Object.keys(updateBody).every((k) => {
+      return allowedUpdate.includes(k);
+    });
+
+    if (!isUpdateAllowed) {
+      throw new Error("update is not allowed");
+    } else {
+      const updatedUser = await User.findByIdAndUpdate(userID, updateBody, {
+        returnDocument: "after",
+        runValidators: true,
+      });
+      if (!updatedUser) {
+        res.status(404).send("User not found");
+      } else {
+        res.send(updatedUser);
+      }
+    }
+  } catch (err) {
+    res.status(400).send({ message: err.message });
```

Comment:
it should be `res.status(400).json({ message: err.message });`

the res.send would send the content type as plain/text which is wrong in case if sending json